### PR TITLE
WIP feat(Add new contact dialog):

### PR DESCRIPTION
**Pull request by daniel7764**
-Added a button under contacts and exposures tab that opens the new dialog
-Dialog allows user to map out all the places the patient was

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import CloseIcon from '@material-ui/icons/Close';
+import {
+    Dialog, DialogTitle, DialogActions, Button, DialogContent,
+    FormControl, Typography, IconButton, Select, MenuItem, Divider, TextField
+} from "@material-ui/core";
+
+import useStyles from './NewContactDialogStyles';
+import Location from 'models/LocationType';
```
Comment (yaniryos) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:9]: סדר IMPORTS

File: client/src/components/App/Content/InvestigationForm/TabManagement/TabManagement.tsx [hunk 2]
```diff
@@ -72,7 +67,7 @@ const TabManagement: React.FC<Props> = (tabManagementProps: Props): JSX.Element
                 </Tabs>
             </Paper>
             {
-                tabs.map((tab) => (
+                tabs.map(tab => (
```
Comment (yaniryos) [inline client/src/components/App/Content/InvestigationForm/TabManagement/TabManagement.tsx:70]: להוסיף TYPE

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx (hunk 1 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:9]: כמובן מה זה CR בלי סדר importים

Comment (yaniryos) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:9]: חחחחחחחחחח

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                                <AddCircleIcon />
+                            </IconButton>
+                            <Typography>{ADD_CONTACT}</Typography>
+                        </div>
+                    </FormControl>
+                </DialogContent>
+                <DialogActions>
+                    <div className={classes.contactCreationActions}>
+                        <Button onClick={() => newContactDialogProps.setDialogOpen(false)}>{CANCEL_LABEL}</Button>
+                        <Button disabled={isCreateContactEnabled}>{ADD_CONTACT}</Button>
```
Comment (RoeiRom) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:91]: Set an empty onclick

File: client/src/components/App/Content/InvestigationForm/TabManagement/TabManagement.tsx [hunk 4]
```diff
@@ -19,19 +18,19 @@ export const tabs: TabObj[] = [
...
-        displayComponent: <></>,
+        displayComponent: <></>
     },
     {
         id: 3,
-        name: 'מקומות ומגעים', 
-        displayComponent: <InteractionsTab/>,
+        name: 'מקומות ומגעים',
         isDisabled: false,
+        displayComponent: <ContactsAndExposuresTab />
```
Comment (RoeiRom) [inline client/src/components/App/Content/InvestigationForm/TabManagement/TabManagement.tsx:33]: This is not the proper component, you need to use Maya’s InteractionsTab component and nest your component inside it

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx [hunk 5]
```diff
@@ -1,65 +1,41 @@
...
 export const END_INVESTIGATION = 'סיים חקירה';
 export const CONTINUE_TO_NEXT_TAB = 'המשך לשלב הבא';
 
 const InvestigationForm: React.FC = (): JSX.Element => {
     const classes = useStyles({});
-    const [exposureDate, setExposureDate] = React.useState<Date>();
-    const [symptomsStartDate, setSymptomsStartDate] = React.useState<Date>();
-    const [hasSymptoms, setHasSymptoms] = React.useState<boolean>(false);
-    const [endInvestigationDate, setEndInvestigationDate] = React.useState<Date>();
+
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx:15]: lines 16-19 suppose to be there

File: client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx [hunk 6]
```diff
@@ -1,65 +1,41 @@
...
-                        </Button>
-                    </div>
+            <InvestigationInfoBar/>
+            <div className={classes.interactiveForm}>
+                <TabManagement
+                    currentTab={currentTab}
+                    setCurrentTab={setCurrentTab}
+                />
+                <div className={classes.buttonSection}>
+                    <Button variant='contained' className={classes.finishInvestigationButton} onClick={() => {
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx:31]: The  color='primary' prop needs to be here - I think you can just replace it here

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/ContactsAndExposureTab.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    {
+        name: 'חנות'
+    },
+    {
+        name: 'קניון'
+    }
+];
+
+export const ADD_CONTACT = 'צור מקום/מגע חדש';
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/ContactsAndExposureTab.tsx:19]: why did you export it? In addition use a more meaningfull name

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { Close as CloseIcon, AddCircle as AddCircleIcon} from '@material-ui/icons';
+import {
+    Dialog, DialogTitle, DialogActions, Button, DialogContent,
+    FormControl, Typography, IconButton, Select, MenuItem, Divider, TextField
+} from "@material-ui/core";
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:6]: use '' here

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        setContactedPersonPhone(event.target.value as string);
+    }
+
+    const handleIDChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+        setContactedPersonID(event.target.value as string);
+    }
+
+    return (
+        <Dialog open={newContactDialogProps.isDialogOpen} maxWidth={false}>
+            <div className={classes.contactDialog}>
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:46]: Maybe you can DialogTitle here? Don't have to change it, only if it makes css easier

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+                <div className={classes.dialogHeader}>
+                    <Typography className={classes.headerTitle}>{DIALOG_TITLE}</Typography>
+                    <IconButton onClick={() => newContactDialogProps.setDialogOpen(false)}>
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:49]: spread all the props to make the use shorter

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.test.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+describe('Contact Addition Card', () => {
+   describe('Field Tests', () => {
+      it('Filling the name field should make the phone field required', () => {});
+      describe('Field Validation Tests', () => {
+          describe('ID Field tests', () => {
+              it('ID field cant contain letters', () => {});
```
[the remark this answers is not in the document]
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.test.tsx:8]: אמרתי שהטסטים עוד לא מוכנים בגלל זה זה בwip

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx (hunk 9 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialog.tsx:46]: DialogTitle wraps the elements inside it with margins and spaces, i preferred using a div in order that ot would seem more like the mockup

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialogStyles.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        flexDirection: 'row',
+        marginRight: '2vh',
+    },
+    fieldLabel: {
+      marginRight: '1vw',
+    },
+    contactCreationActions: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: '',
```
Comment (EmilyLip) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactsAndExposurePlacesTab/NewContactDialog/NewContactDialogStyles.tsx:48]: remove if unnecessary

**Pull request closed without merging**